api/report_server.thrift

```diff
@@ -303,9 +306,44 @@ service codeCheckerDBAccess {
   // Analysis result storage related API calls.
   //============================================
 
-  // store checker run related data to the database
-  // by default updates the results if name was already found
-  // using the force flag removes existing analysis results for a run
+  // !!!WARNING!!! The store API has a very strict way of usage! The best if
+  // one calls the storeZip() function which handles the storage of a full run.
+  // See the documentation of the storeZip() function.
+  // The other storage functions can also be used to store reports. The calling
+  // order must be:
+  // 1. addFileContent()
+  // 2. addCheckerRun()
+  // 3. addReport(), addSkipPath(), setRunDuration()
+  // 4. finishCheckerRun()
+
+  // The client can ask the server whether a file is already stored in the
+  // database. If it is, then it is not necessary to send it in the ZIP file
+  // with storeZip() function. This function requires a list of file hashes
+  // (sha256) and returns the ones which are not stored yet.
+  list<string> necessaryFileContents(
+                                     1: list<string> file_hashes)
+                                     throws (1: shared.RequestFailed requestError),
+
+  // This function stores an entire run encapsulated and sent in a ZIP file.
+  // The ZIP file has to be sent as a base64 encoded string. The ZIP file must
+  // contain exactly one directory which has a "reports" and an optional "root"
+  // sub-folder. The former one is the output of "CodeChecker analyze" command
+  // and the latter one contains the source files on absolute paths starting as
+  // if "root" was the "/" directory. The source files are not necessary to be
+  // wrapped in the ZIP file (see necessaryFileContents() function).
+  i64 storeZip(
```

Comment:
With binary data such as a ZIP file, the `base64` encoding should be better, even if that Thrift bug gets fixed.